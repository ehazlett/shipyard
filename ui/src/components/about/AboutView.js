import React from 'react';

import { Grid, Menu } from 'semantic-ui-react';

import { getVersion, getInfo } from '../../api';


class SettingsView extends React.Component {
  state = {
    info: null,
    version: null,
    error: null,
    activeSegment: 'info',
    loading: true,
  };

  componentDidMount() {
    this.getSwarmInfo();
    this.getSwarmVersion();
  }

  getSwarmVersion = () => {
    getVersion()
      .then((version) => {
        this.setState({
          error: null,
          version: version.body,
        });
      })
      .catch((error) => {
        this.setState({
          error,
        });
      });
  };

  getSwarmInfo = () => {
    getInfo()
      .then((info) => {
        this.setState({
          error: null,
          info: info.body,
          loading: false,
        });
      })
      .catch((error) => {
        this.setState({
          error,
          loading: false,
        });
      });
  };

  changeSegment = (name) => {
    this.setState({
      activeSegment: name,
    });
  };

  render() {
    var { version, info, loading, activeSegment } = this.state;
    console.log(version, info);

    if(loading) {
      return <div></div>;
    }

    return (
      <Grid padded>
        <Grid.Column width={16}>
          <Menu pointing secondary>
            <Menu.Item name='Info' active={activeSegment === 'info'} onClick={() => { this.changeSegment('info'); }} />
            <Menu.Item name='Version' active={activeSegment === 'version'} onClick={() => { this.changeSegment('version'); }} />
          </Menu>

          <pre>{ activeSegment === 'info' ? JSON.stringify(info, null, '  ') : null }{ activeSegment === 'version' ? JSON.stringify(version, null, '  ') : null }</pre>
        </Grid.Column>
      </Grid>
    );
  }
}

export default SettingsView;
