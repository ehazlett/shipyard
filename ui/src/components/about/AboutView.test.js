import React from "react";
import ReactDOM from "react-dom";
import AboutView from "./AboutView";
import renderer from "react-test-renderer";

it("AboutView renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<AboutView />, div);
});

it("AboutView snapshot matches", () => {
  const tree = renderer.create(<AboutView />).toJSON();
  expect(tree).toMatchSnapshot();
});
