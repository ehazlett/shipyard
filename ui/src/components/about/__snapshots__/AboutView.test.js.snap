exports[`test AboutView snapshot matches 1`] = `<div />`;
